pyfar/dsp/normalize.py

```diff
@@ -0,0 +1,151 @@
+    """
+    Parameters
+    ----------
+    input_signal:
+        signal of the signal class
+        
+    norm_type:
+        'time' - normalizes time signal to 'value' 
+        'abs' - normalizes magnitude spectrum to 'value'
+        'dB' - normalizes db magnitude spectrum to 'value'
+        (default = 'time')
+
+    operation:
+        'max' - finds absolute maximum of data to normalize 
+        'mean' - find the mean of data to normalize
+        'rms' - fins the rms of data to normalize
+        (default = 'max')
+    
+    channelwise:
+        'each' - normalize each channel seperately 
+        'max' - normalize to max or operation across all channels
+        'min' - normalize to min or operation across all channels
+        'mean' - normalize to mean of operation across all channels
+        (default = 'max')
+    
+    value: 
+        normalizes to 'value' which can be a scalor or a vector 
+        with a number of elements equal to channels. Default is 0 
+        for type = 'db' and 1 otherwise
+    f:
+        two element vector specifiying upper and lower frequenzy bounds
+        for normalization or scalar specifying the centre frequency for 
+        normalization (see 'frac', default is [0,fs/2])
+    fs:
+        sampling frequency in Hz (default = 44100)
+        
+    Returns
+    --------
+    signal_normalized - normalized input signal
+    """
+    
+    # set defaults
+    
+    if value == 'unnasigned':
+        if norm_type == 'dB':
+            value = 0
+        else:
+            value = 1
+    
+    # copy input
+    
+    input_normalized = input_signal.time.copy()
+
+    # flatten
+    
+    #input_normalized =  input_normalized.flatten() # Flattening / Reshaping to a 2d array to work with possible?
+
+    # transform data to the desired domain
+    
+    if norm_type == 'time':
+         input_normalized = input_normalized
+    
+    elif norm_type == 'abs':
+        input_normalized = np.abs(np.fft.fft(input_normalized,input_normalized.ndim - 1)) 
+    
+    elif norm_type == 'dB':
+        input_normalized = np.fft.fft(input_normalized,input_normalized.ndim - 1)
+        
+    else:
+        raise ValueError(("norm _type must be 'time', 'abs' or 'dB'"))
+      
+    # get bounds for normalization
+    
+    if norm_type == 'time':
+        lim = np.array([0, input_normalized.shape[1]])
+    
+    else: # if domain is abs or db
+        
+        if len(f) == 1:
+            #frequency spacing
+            df = fs/(input_normalized.shape[1])
+            
+            # get frequency limits
+            f = f/2^(1/2)
+            f = np.array([f, f*2])
+            
+            # get corresponding indicees
+            lim = np.round(f/df) + 1
+            
+        else: 
+            # frequency spacing
+            df = fs/(input_normalized.shape[1])
+            
+            # get corresponding indices
+            lim = np.round(f/df) + 1
```

Comment:
This can be done with the Signal class method:
`lim = signal.find_nearest_frequency(f)`

Because the value at 0 Hz and Nyquist depend on the FFT normalization, I would suggest to exclude them:
```
if signal.n_samples % 2:
    lim[0] = np.max([lim[0, 1])
else:
    lim = np.clip(lim, 0)
```